### DISCUSSION TITLE
Restoring compatibility with Vue 3 component library

### DISCUSSION THREAD

**Discussion by AlessandroAmos** [beholdr/maska#139] [Ideas]
In version 1, when using the v-mask directive, even if the v-mask directive was used in input components from famous Vue 3 libraries such as Vuetify and NaiveUI, the mask worked perfectly... I would like this to be made available in this new version as well.

**Comment by beholdr** [comment]
It should work now too. Could you please provide online example for me to see?

**Comment by radimhornicek** [comment]
I can confirm that behavior. In nuxt v3 worked directive beholder-maska@1.x with Vuetify@3.x side by side. Current v2.1.7 not. In console - there is no error. It just doesnt work anymore. I checked component in DevVue and it seems that Vuetify pass down all attr including data-maska (as well as directive I belive)

**Reply by beholdr** [reply]
Here is [example](https://play.vuetifyjs.com/#eNqVUu1OwjAUfZWm/JhE1vKl6AIGH8D4AIwfcytQWbum7aaG7N297QYZKCYmy7bTc3vOub1dHfCzUqQqGY7w3DKh8sSyp1giNK/CRCn/64FIuGyRx5Z92nDDWZ4hIIuM5YsYC7ONscOJ2SfRyn9eleWFNGtEj2K0o+ZA4zOnHX+AJtVcWWSYLR3NhSq0RQek2QbVaKMLgQIIHnSo6sUZnlhvD3wsUwhgEaRDC7f/JhiOxpPp3f3s4THon+hOWqg7uHhuLULB7Qjd9Hq9PoJX6B+QrV3mJiXkwwPc5IDeFHk3hYQj9RpxS5gYR42qW4PoDsd4Z60yEaWlVPstSQtBl8BRXUrLBQuzQiwnZEKGYZKrXUKmNOPGdmnCjAjfdPFhmAbjGA86HpZvvq75tDTIj8iskW2XnOQPKQr6FdOhZjJjmuk/41/U/tbCRcn1Nvxgzt3STEINXDpeaSKZpVIJ6uuW40bdA6/jZGBWNUwo53JvLoaTGj+Y1dHtX+ckuCRewO1eO5N6gM96BVe/D6+/ATX5IcA=) with Vuetify 3.1.7

**Reply by radimhornicek** [reply]
Hm interisting, I tried that only with data-maska attribue eg. data-maska="+1 (###) ###-##-##" and that doesn't work. Maybe I just didn't understand how it works. Your solution is good enough. Thank you for your help!

**Reply by beholdr** [reply]
Problem is vuetify does not pass `data-maska` attribute to native input so you need to use directive options.

**Reply by inNetto** [reply]
```
const maskaOptions = {
  mask: '+1 (###) ###-##-##',
  onMaska: (detail: MaskaDetail) => console.log(detail.unmasked)
}

<v-text-field
        v-model="cpf"
        v-maska:[maskaOptions]
        label="Cpf"
        :error-messages="errors.cpf"
/>


```
Worked. Thanks
